src/System.Text.Json/tests/WritableJsonApiTests.cs

```diff
@@ -120,6 +153,20 @@ public static void TestCreatingJsonObject()
                 { "is developer", true },
                 { "null property", (JsonNode) null }
             };
+
+            Assert.IsType<JsonString>(developer["name"]);
+            var developerNameCasted = developer["name"] as JsonString;
+            Assert.Equal("Kasia", developerNameCasted.Value);
+
+            Assert.IsType<JsonNumber>(developer["age"]);
+            var developerAgeCasted = developer["age"] as JsonNumber;
+            Assert.Equal(22, developerAgeCasted.GetInt32());
```

Comment:
Just wondering, Value also has a public Set right? which means that I should be able to mutate the age if I wanted to?

Reply:
Yes, as for example in: `TestModifyingJsonObjectPrimaryTypes`